onnxruntime/core/session/inference_session.cc

```diff
@@ -528,6 +528,16 @@ common::Status InferenceSession::Initialize() {
     // now that all the transforms are done, call Resolve on the main graph. this will recurse into the subgraphs.
     ORT_RETURN_IF_ERROR(graph.Resolve());
 
+    if (!session_options_.optimized_model_filepath.empty()) {
+      if (session_options_.graph_optimization_level < TransformerLevel::Level3) {
+        // Serialize optimized ONNX model.
+        ORT_RETURN_IF_ERROR(Model::Save(*model_, session_options_.optimized_model_filepath));
+      } else {
+        LOGS(*session_logger_, ERROR) << "Serializing Optimized ONNX model with Graph Optimization"
```

Comment:
Please use warning instead of error. Since this is not a on recoverable error and we continue execution after logging this so the appropriate level for this is warning.